src/hotspot/share/gc/shenandoah/heuristics/shenandoahOldHeuristics.cpp

```diff
@@ -310,37 +309,29 @@ void ShenandoahOldHeuristics::prepare_for_old_collections() {
 
   const size_t num_regions = heap->num_regions();
   size_t cand_idx = 0;
-  size_t total_garbage = 0;
   size_t immediate_garbage = 0;
   size_t immediate_regions = 0;
   size_t live_data = 0;
 
   RegionData* candidates = _region_data;
   for (size_t i = 0; i < num_regions; i++) {
     ShenandoahHeapRegion* region = heap->get_region(i);
-    if (!_old_generation->contains(region)) {
+    if (!region->is_old()) {
       continue;
     }
 
     size_t garbage = region->garbage();
     size_t live_bytes = region->get_live_data_bytes();
-    total_garbage += garbage;
     live_data += live_bytes;
 
-    // Only place regular regions into the candidate set
-    if (region->is_regular()) {
-      if (!region->has_live()) {
-        assert(!region->is_pinned(), "Pinned region should have live (pinned) objects.");
-        region->make_trash_immediate();
-        immediate_regions++;
-        immediate_garbage += garbage;
-      } else {
-        region->begin_preemptible_coalesce_and_fill();
-        candidates[cand_idx]._region = region;
-        candidates[cand_idx]._u._live_data = live_bytes;
-        cand_idx++;
-      }
+    if (region->is_trash()) {
```

Comment:
In revised code structure, we handle trash regular regions with same code that handles trash humongous regions.  I like this improvement.  This is a note to self.  Please let me know if I am misunderstaning.

Reply:
Need to double-check myself here.  is_trash() may not be exactly the same as !region->has_live().  If these are distinct, maybe we want to check for either condition right here.

Reply:
I see that we need to explicitly make_trash() after we discover that !region->has_live().  So these are different conditions.